static/js/import_wizard/components/value_map_section.tsx

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for showing and updating the mapping of cell values.
+ */
+
+import _ from "lodash";
+import Papa from "papaparse";
+import React, { useEffect, useState } from "react";
+import { Button, Input } from "reactstrap";
+
+import { ValueMap } from "../types";
+
+interface ValueMapSectionProps {
+  valueMap: ValueMap;
+  onValueMapSubmitted: (valueMap: ValueMap) => void;
+  onValueMapUpdated: () => void;
+}
+
+interface ValueMapInput {
+  originalVal: string;
+  newVal: string;
+}
+
+const EMPTY_VALUE_MAP_INPUT = {
+  originalVal: "",
+  newVal: "",
+};
+
+export function ValueMapSection(props: ValueMapSectionProps): JSX.Element {
+  const [uploadedFile, setUploadedFile] = useState(null);
+  const [mappingInput, setMappingInput] = useState(
+    getMappingInput(props.valueMap)
+  );
+  const [showFileUpload, setShowFileUpload] = useState(false);
+  const [showConfirmationButton, setShowConfirmationButton] = useState(true);
+  const mappingInputOrderedList = Object.keys(mappingInput).sort(
+    (a, b) => Number(a) - Number(b)
+  );
+
+  useEffect(() => {
+    props.onValueMapUpdated();
+    setShowConfirmationButton(true);
+  }, [uploadedFile, mappingInput]);
+
+  return (
+    <>
+      <div className="section-container">
+        <div className="section-header">
+          <h2>Value Mappings</h2>
+          <div className="header-tab-section">
+            <div
```

Comment:
(for later) We could use a toggle switch buttons here like we do for scatter

Reply:
done, added TODO